tests/kernel/kernel_parameters.cpp

```diff
@@ -0,0 +1,441 @@
+                                        "sycl::vec<" + primary_type_name +
+                                            ", 4>",
+                                        primary_type_name + "[ARRAY_SIZE]",
+                                        "Base<" + primary_type_name + ">",
+                                        "Derived<" + primary_type_name + ">");
+    return types;
+  } else {
+    static const auto types =
+        named_type_pack<PrimaryType[ARRAY_SIZE], Base<PrimaryType>,
+                        Derived<PrimaryType>>::generate(primary_type_name +
+                                                            "[ARRAY_SIZE]",
+                                                        "Base<" +
+                                                            primary_type_name +
+                                                            ">",
+                                                        "Derived<" +
+                                                            primary_type_name +
+                                                            ">");
+    return types;
+  }
+}
+
+inline auto get_ranges_type_pack() {
+  static const auto types =
+      named_type_pack<sycl::range<1>, sycl::range<2>, sycl::range<3>>::generate(
+          "sycl::range<1>", "sycl::range<2>", "sycl::range<3>");
+  return types;
+}
+
+inline auto get_ids_type_pack() {
+  static const auto types =
+      named_type_pack<sycl::id<1>, sycl::id<2>, sycl::id<3>>::generate(
+          "sycl::id<1>", "sycl::id<2>", "sycl::id<3>");
+  return types;
+}
+
+template <typename T>
+void init_data(T& data, int value) {
+  if constexpr (std::is_array_v<T>) {
+    std::remove_reference_t<decltype(data[0])> tmp;
+    for (size_t i = 0; i < ARRAY_SIZE; ++i) {
+      data[i] = value_operations::init<decltype(tmp)>(value);
+    }
+  } else {
+    data = value_operations::init<T>(value);
+  }
+}
+
+template <typename AccType, typename KerParDerivedType>
+class named_kernel {
+  KerParDerivedType ker_par;
+  AccType buf_acc;
+
+ public:
+  named_kernel(AccType buf_acc) {
+    this->buf_acc = buf_acc;
+    init_data(ker_par, changed_val);
+  }
+  void operator()() const {
+    if constexpr (!std::is_array_v<KerParDerivedType>)
+      buf_acc[0] = ker_par;
+    else {
+      for (size_t i = 0; i < ARRAY_SIZE; ++i) buf_acc[i] = ker_par[i];
+    }
+  }
+};
+
+template <typename T>
+class named_kernel_test {
+  template <typename BufType>
+  void queue_submit_task(BufType buf_expected) {
+    queue
+        .submit([&](sycl::handler& cgh) {
+          auto acc_expected =
+              buf_expected.template get_access<sycl::access_mode::read_write>(
+                  cgh);
+          named_kernel<decltype(acc_expected), T> kernel(acc_expected);
+          cgh.single_task(kernel);
+        })
+        .wait_and_throw();
+  }
+  sycl::queue queue;
+
+ public:
+  named_kernel_test() { auto queue = sycl_cts::util::get_cts_object::queue(); }
```

Comment:
I am confused about this line which appears many times. What is it for?

Reply:
Fixed.